### ISSUE TITLE
0.7-beta Resource isnt updated with source signal

### ISSUE THREAD

**Issue by VVishion** [leptos-rs/leptos#2716]
**Describe the bug**

`Resource` isnt updated with source signal

```rust
async fn create_list(i: usize) -> Vec<String> {
    if i % 2 == 0 {
        log!("create even list");
        vec![
            String::from("Hello,"),
            String::from("Dave"),
            String::from("and"),
            String::from("the"),
            String::from("green"),
            String::from("World"),
        ]
    } else {
        log!("create uneven list");
        vec![String::from("Ehm")]
    }
}

#[component]
pub fn App() -> impl IntoView {
    let (counter, mut_counter) = signal(0);

    let list = Resource::new(move || counter.get(), create_list);

    Effect::new(move |_| {
        log!("{:?}", list.get());
    });

    Effect::new(move |_| {
        log!("{:?}", counter.get());
    });

    view! {
        <div id="app">
            {move || counter.get()}
            <button on:click=move |_| {
                log!("click");
                mut_counter.update(|counter| {
                    *counter += 1;
                });
            }>"Click here!"</button>

            {move || list.get()}
        </div>
    }
}
```

**Leptos Dependencies**

```toml
leptos = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
leptos_axum = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
leptos_meta = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
leptos_router = { git = "https://github.com/leptos-rs/leptos.git", branch = "leptos_0.7" }
```

at the time of commit 4bc999ead36ab9eeb2b80e13ef945b343e9e3393

**To Reproduce**
Click the button. The list doesnt change.
On the first click the fetcher refires, but the counter passed into the fetcher is not incremented, yet.
All subsequent clicks it doesnt refetch.

**Expected behavior**
When clicking the button, it should toggle between "Ehm" and "Hello,DaveandthegreenWorld"

**Additional Context**
Take a look at the console logs

**Comment by VVishion** [comment]
I think a better example is
```rust
#[server]
pub async fn create_server_list(i: usize) -> Result<Vec<String>, ServerFnError> {
    if i % 2 == 0 {
        log!("create even list");
        Ok(vec![
            String::from("Hello,"),
            String::from("Dave"),
            String::from("and"),
            String::from("the"),
            String::from("green"),
            String::from("World"),
        ])
    } else {
        log!("create uneven list");
        Ok(vec![String::from("Ehm")])
    }
}

#[component]
pub fn App() -> impl IntoView {
    let (counter, mut_counter) = signal(0);

    //let list = Resource::new(move || counter.get(), create_server_list);
    let list = LocalResource::new(move || async move { create_server_list(counter.get()).await });

    Effect::new(move |_| {
        log!("{:?}", list.get());
    });

    Effect::new(move |_| {
        log!("{:?}", counter.get());
    });

    view! {
        <div id="app">
            {move || counter.get()}
            <button on:click=move |_| {
                log!("click");
                mut_counter.update(|counter| {
                    *counter += 1;
                });
            }>"Click here!"</button>

            <Suspense>
                {move || list.get()}
            </Suspense>
        </div>
    }
}
```
Where LocalResource works, but Resource doesnt

**Comment by gbj** [comment]
Thanks, this is a good issue to find. I've been investigating it today and it's a bit tricky, but I'll figure it out. Strangely, the source signals for examples like the `todo_app_sqlite` ones do work. It looks like the effect is triggering twice per change, which may be part of it. (I think the same issue may drive #2721 but we'll see.)

**Comment by gbj** [comment]
I believe this should be fixed by 5d09cd1, but I don't think #2721 is fixed yet.
